src/vs/platform/files/test/browser/indexedDBFileService.test.ts

```diff
@@ -45,33 +72,246 @@ suite('IndexedDB File Service', function () {
+	}
+
+	test('deleteFile', async () => {
+		await initFixtures();
+
+		let event: FileOperationEvent;
+		disposables.add(service.onDidRunOperation(e => event = e));
+
+		const anotherResource = userdataURIFromPaths(['fixtures', 'service', 'deep', 'company.js']);
+		const resource = userdataURIFromPaths(['fixtures', 'service', 'deep', 'conway.js']);
+		const source = await service.resolve(resource);
+
+		assert.equal(await service.canDelete(source.resource, { useTrash: false }), true);
+		await service.del(source.resource, { useTrash: false });
+
+		assert.equal(await service.exists(source.resource), false);
+		assert.equal(await service.exists(anotherResource), true);
+
+		assert.ok(event!);
+		assert.equal(event!.resource.path, resource.path);
+		assert.equal(event!.operation, FileOperation.DELETE);
+
+		{
+			let error: Error | undefined = undefined;
+			try {
+				await service.del(source.resource, { useTrash: false });
+			} catch (e) {
+				error = e;
+			}
+
+			assert.ok(error);
+			assert.equal((<FileOperationError>error).fileOperationResult, FileOperationResult.FILE_NOT_FOUND);
+		}
+		await reload();
+		{
+			let error: Error | undefined = undefined;
+			try {
+				await service.del(source.resource, { useTrash: false });
+			} catch (e) {
+				error = e;
+			}
+
+			assert.ok(error);
+			assert.equal((<FileOperationError>error).fileOperationResult, FileOperationResult.FILE_NOT_FOUND);
+		}
+	});
+
+	test('deleteFolder (recursive)', async () => {
+		await initFixtures();
+		let event: FileOperationEvent;
+		disposables.add(service.onDidRunOperation(e => event = e));
+
+		const resource = userdataURIFromPaths(['fixtures', 'service', 'deep']);
+		const subResource1 = userdataURIFromPaths(['fixtures', 'service', 'deep', 'company.js']);
+		const subResource2 = userdataURIFromPaths(['fixtures', 'service', 'deep', 'conway.js']);
+		assert.equal(await service.exists(subResource1), true);
+		assert.equal(await service.exists(subResource2), true);
+
+		const source = await service.resolve(resource);
+
+		assert.equal(await service.canDelete(source.resource, { recursive: true, useTrash: false }), true);
+		await service.del(source.resource, { recursive: true, useTrash: false });
+
+		assert.equal(await service.exists(source.resource), false);
+		assert.equal(await service.exists(subResource1), false);
+		assert.equal(await service.exists(subResource2), false);
+		assert.ok(event!);
+		assert.equal(event!.resource.fsPath, resource.fsPath);
+		assert.equal(event!.operation, FileOperation.DELETE);
+	});
+
+
+	test('deleteFolder (non recursive)', async () => {
+		await initFixtures();
+		const resource = userdataURIFromPaths(['fixtures', 'service', 'deep']);
+		const source = await service.resolve(resource);
+
+		assert.ok((await service.canDelete(source.resource)) instanceof Error);
+
+		let error;
+		try {
+			await service.del(source.resource);
+		} catch (e) {
+			error = e;
+		}
+		assert.ok(error);
+	});
```

Comment:
Added a timing-based test of batching, driven by the principle that without batching is ~20x slower and would timeout causing an error to be thrown. This may be relying too much on the state of the testing machine, so maybe it'd be better to keep some sort of count of IDB requests fired and expose that for testing? What do you think?